instrumentation/opentelemetry-instrumentation-openai/src/opentelemetry/instrumentation/openai_v2/utils.py

```diff
@@ -0,0 +1,162 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+from typing import Optional, Union
+
+from openai import NOT_GIVEN
+
+from opentelemetry.semconv._incubating.attributes import (
+    gen_ai_attributes as GenAIAttributes,
+)
+from opentelemetry.trace import Span
+
+
+def silently_fail(func):
+    """
+    A decorator that catches exceptions thrown by the decorated function and logs them as warnings.
+    """
+
+    logger = logging.getLogger(func.__module__)
+
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as exception:
+            logger.warning(
+                "Failed to execute %s, error: %s",
+                func.__name__,
+                str(exception),
+            )
+
+    return wrapper
+
+
+def extract_content(choice):
+    if getattr(choice, "message", None) is None:
+        return ""
+
+    # Check if choice.message exists and has a content attribute
+    message = choice.message
+    if getattr(message, "content", None):
+        return choice.message.content
+
+    # Check if choice.message has tool_calls and extract information accordingly
+    elif getattr(message, "tool_calls", None):
+        result = [
+            {
+                "id": tool_call.id,
+                "type": tool_call.type,
+                "function": {
+                    "name": tool_call.function.name,
+                    "arguments": tool_call.function.arguments,
+                },
+            }
+            for tool_call in choice.message.tool_calls
+        ]
+        return result
+
+    # Check if choice.message has a function_call and extract information accordingly
+    elif getattr(message, "function_call", None):
+        return {
+            "name": choice.message.function_call.name,
+            "arguments": choice.message.function_call.arguments,
+        }
+
+    # Return an empty string if none of the above conditions are met
+    else:
+        return ""
+
+
+def extract_tools_prompt(item):
+    tool_calls = getattr(item, "tool_calls", None)
+    if tool_calls is None:
+        return
+
+    calls = []
+    for tool_call in tool_calls:
+        tool_call_dict = {
+            "id": getattr(tool_call, "id", ""),
+            "type": getattr(tool_call, "type", ""),
+        }
+
+        if hasattr(tool_call, "function"):
+            tool_call_dict["function"] = {
+                "name": getattr(tool_call.function, "name", ""),
+                "arguments": getattr(tool_call.function, "arguments", ""),
+            }
+        calls.append(tool_call_dict)
+    return calls
+
+
+def set_event_prompt(span: Span, prompt):
```

Comment:
Nit: No point in having typing annotations if we don't have it for the whole function :)